src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/config/DataSourceElement.java

```diff
@@ -288,6 +291,16 @@ private BasicDataSource initPool(String maxPool) {
             dataSource.setPassword(getPassword());
         }
 
+        if(isPreinit()) {
+            // side effect - connection pool init - that is what we want
+            try {
+                dataSource.getConnection().close();
+                log.debug("Preinitializing the connection pool: {}@{}", getDataSourceName(), System.identityHashCode(dataSource));
```

Comment:
can you surround it with log.isDebugEnabled() ?

Reply:
Hi, I found a lot of examples where calls to log.debug() are not surrounded with log.isDebugEnabled(). Others are, though...  When to use which approach?

Reply:
Technically speaking, `isDebugEnabled` should be used when "arguments have computations" or "varargs are used for logging".

In this case, `System.identityHashCode` will be computed no matter if `debug` is enabled or not, so it it makes sense to use `isDebugEnabled.

Even though this "connection pool initialization" will be used just once, the code might evolve (or even be copy-pasted"), so I think it is worth adding `isDebugEnabled` when the first statement holds.

Reply:
thanks, I surrounded it with a check